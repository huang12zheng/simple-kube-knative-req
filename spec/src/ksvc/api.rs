//! In the end, [] ref to [GVKSpecWrapper] and [BuilderApi].
//!
//! We need From [KnativeSpec] Into [GVKSpecWrapper],
//!
//! LOOK ALSO:
//! - [src/gvk/api.md]
//! - [BuilderApi]

use crate::*;

#[async_trait]
impl GetApi for KnativeSpec {
    async fn get_api(&self) -> Api<DynamicObject> {
        let gvk = self.into_gvk();
        return gvk.get_api().await;
    }
}

#[cfg(test)]
mod test {
    use crate::{GetApi, KnativeSpec};

    // #[cfg(test)]
    // use crate::ksvc_into_gvk;

    #[tokio::test]
    async fn show_api() {
        let spec = KnativeSpec::default();
        let api = spec.get_api().await;
        insta::assert_debug_snapshot!(api);
    }
}
