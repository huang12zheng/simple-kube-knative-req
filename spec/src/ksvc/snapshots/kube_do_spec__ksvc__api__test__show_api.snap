---
source: spec/src/ksvc/api.rs
expression: api
---
Api {
    request: Request {
        url_path: "/apis/serving.knative.dev/v1/namespaces/default/services",
    },
    client: "...",
    namespace: Some(
        "default",
    ),
}
